ledger/src/blockstore_processor.rs

```diff
@@ -921,21 +920,44 @@ pub fn confirm_slot(
 
     let verify_transaction = {
         let bank = bank.clone();
-        move |versioned_tx: VersionedTransaction| -> Result<SanitizedTransaction> {
-            bank.verify_transaction(versioned_tx, skip_verification)
+        move |versioned_tx: VersionedTransaction,
+              skip_verification: bool,
+              verify_precompiles: bool|
```

Comment:
what's the point of separating these two? When would we verify precompiles but not verify everything?

Reply:
It's really to enable the logic of the start_verify_transactions to do everything except the sigverify, which is done on the GPU. So when for example Cuda is enabled, start_verify_transactions will do something like

    let verify_func = {
        move |versioned_tx: VersionedTransaction| -> Result<SanitizedTransaction> {
            verify(versioned_tx, true, true)
        }
    };
    let entries = verify_transactions(entries, Arc::new(verify_func));
    // do GPU sigverify

so that we skip the (CPU-side) sigverify, but still do verify_precompiles. After this, we do the sigverify on the GPU.

Reply:
Cool, thanks for the explanation! I ended up realizing that after posting that comment. I would suggest adding a new `VerificationOptions` struct with named fields for each thing we are verifying. `bool`'s can be a little hard to track